src/battle/battle_command.c

```diff
@@ -1198,7 +1198,7 @@ BOOL BtlCmd_CalcExpGain(BattleSystem *bsys, BattleContext *ctx) {
         Pokemon *mon;
         for (int i = 0; i < Party_GetCount(BattleSystem_GetParty(bsys, 0)); i++) {
             mon = BattleSystem_GetPartyMon(bsys, 0, i);
-            if (GetMonData(mon, MON_DATA_SPECIES, 0) && GetMonData(mon, MON_DATA_HP, 0)) {
+            if (GetMonData(mon, MON_DATA_SPECIES, 0) && GetMonData(mon, MON_DATA_CUR_HP, 0)) {
```

Comment:
not 100% sure if the CUR is necessary
I think people would know kinda what MON_DATA_HP means
either way it doesn't really make a difference
not something that needs to be changed if you feel strongly about it

Reply:
same applies to PP